src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1404,6 +1606,37 @@ final case class TStruct(fields: IndexedSeq[Field]) extends Type {
       }))
   }
 
+  override def unsafeOrdering(missingGreatest: Boolean): UnsafeOrdering = {
+    val mg = missingGreatest
+    val fieldOrderings = fields.map(_.typ.unsafeOrdering(missingGreatest)).toArray
+
+    // FIXME: remove when genome reference comes in and we can compare fundamental types
+    val repr = fundamentalType
```

Comment:
could actually add fundamentalType to the loadField method. That would solve it.